merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/ConstraintAction.java

```diff
@@ -152,30 +147,58 @@ public List<ConstraintResult> getViolations(final PlanId planId, final Optional<
-            plan.missionModelId,
-            Optional.of(planId),
-            Optional.of(simDatasetId),
-            constraint.definition()
-        );
+        ConstraintsDSLCompilationService.ConstraintsDSLCompilationResult constraintCompilationResult = null;
+        try {
+          constraintCompilationResult = constraintsDSLCompilationService.compileConstraintsDSL(
+              plan.missionModelId,
+              Optional.of(planId),
+              Optional.of(simDatasetId),
+              constraint.definition()
+          );
+        } catch (MissionModelService.NoSuchMissionModelException ex) {
+          constraintResultMap.put(
+              constraint.id(),
+              Failable.failure(new Error("Constraint " + constraint.name() + ": " + ex.getMessage())));
+          continue;
+        } catch (NoSuchPlanException ex) {
+          constraintResultMap.put(
+              constraint.id(),
+              Failable.failure(new Error("Constraint " + constraint.name() + ": " + ex.getMessage())));
+          continue;
+        }
 
+        // Try to compile the constraint and caputure failure
         if (constraintCompilationResult instanceof ConstraintsDSLCompilationService.ConstraintsDSLCompilationResult.Success success) {
           expression = success.constraintExpression();
         } else if (constraintCompilationResult instanceof ConstraintsDSLCompilationService.ConstraintsDSLCompilationResult.Error error) {
-          throw new ConstraintCompilationException(constraint.name(),error);
+          constraintResultMap.put(
+              constraint.id(),
+              Failable.failure(error, "Constraint '" + constraint.name() + "' compilation failed:\n "));
+          continue;
         } else {
-          throw new ConstraintCompilationException(constraint.name(),new ConstraintsDSLCompilationService.ConstraintsDSLCompilationResult.Error(new ArrayList<>(){{
-            add(new ConstraintsCompilationError.UserCodeError(
-                "Unhandled variant of ConstraintsDSLCompilationResult: " + constraintCompilationResult,
-                "",
-                new ConstraintsCompilationError.CodeLocation(0, 0),
-                ""));
-          }}));
+          final var finalConstraintCompilationResult = constraintCompilationResult;
+          constraintResultMap.put(
+              constraint.id(),
+              Failable.failure(
+                  new ConstraintsDSLCompilationService.ConstraintsDSLCompilationResult.Error(
+                      new ArrayList<>() {{
+                        add(new ConstraintsCompilationError.UserCodeError(
+                            "Unhandled variant of ConstraintsDSLCompilationResult: "
+                            + finalConstraintCompilationResult,
+                            "",
+                            new ConstraintsCompilationError.CodeLocation(
+                                0,
+                                0),
+                            ""));
```

Comment:
Minor, ignorable: formatting
```suggestion
                      new ArrayList<>() {{
                        add(new ConstraintsCompilationError.UserCodeError(
                            "Unhandled variant of ConstraintsDSLCompilationResult: " + finalConstraintCompilationResult,
                            "",
                            new ConstraintsCompilationError.CodeLocation(0,0),
                            ""));
```

Reply:
Ignore.